### DISCUSSION TITLE
Nested Groups

### DISCUSSION THREAD

**Discussion by tpaksu** [tailwindlabs/tailwindcss#2564] [Ideas]
Hello, 

I want to create a nested menu like this:

```html
<ul>
    <li class="group">
        <div class='flex px-3 py-2'>
            <i class='icon mr-2 text-gray-300 group-hover:text-yellow-300'></i>
            <span class='text-gray-300 group-hover:text-yellow-300'>text</span>
        </div>
        <ul>
            <li class="group">
                <div class='flex px-3 py-2'>
                    <i class='icon mr-2 text-gray-300 group-hover:text-yellow-300'></i>
                    <span class='text-gray-300 group-hover:text-yellow-300'>text</span>
                </div>
            </li>
        </ul>
    </li>
</ul>
```

When I hover to a menu element in the top-level menu, the dropdown shows with all menu items having yellow colors. Because group-hover clause will take effect on all nested items of the topmost list item. I know I can extend tailwindcss to get what I want, but I wanted to ask what you think about having levels in the default configuration like this:
    
```html
<ul>
    <li class="group">
        <div class='flex px-3 py-2'>
            <i class='icon mr-2 text-gray-300 group-hover:text-yellow-300'></i>
            <span class='text-gray-300 group-hover:text-yellow-300'>text</span>
        </div>
        <ul>
            <li class="group-1">
                <div class='flex px-3 py-2'>
                    <i class='icon mr-2 text-gray-300 group-1-hover:text-yellow-300'></i>
                    <span class='text-gray-300 group-1-hover:text-yellow-300'>text</span>
                </div>
            </li>
        </ul>
    </li>
</ul>
```

This way the groups won't interfere with each other. I think having groups 8 levels deep would cover every case. Or this can be configurable on build too. 

Thanks.

**Comment by tpaksu** [comment]
And yes, I know moving the `group` class to the first div inside will fix the error, but this is just a simple example to demonstrate the problem. Something would be needed without altering the DOM, for example, when you don't have the wrapper div, around the icon and text.

**Comment by reinink** [comment]
Hey! Good news — we just added nested group and multiple peer support using variant modifiers in Tailwind CSS v3.2 🥳

You can read about it in our [announcement blog post](https://tailwindcss.com/blog/tailwindcss-v3-2#nested-group-and-multiple-peer-support-using-variant-modifiers), but for quick reference here’s what it looks like:

```html
<div class="group/sidebar ...">
  <!-- ... -->
  <div class="group/navitem ...">
    <a
      href="#"
      class="opacity-50 group-hover/sidebar:opacity-75 group-hover/navitem:bg-black/75"
    >
      <!-- ... -->
    </a>
  </div>
  <!-- ... -->
</div>
```

This lets you give each group a clear name that makes sense for that context on the fly, and Tailwind will generate the necessary CSS to make it work.
